src/jobs/ipni-advertisement-fetcher-job/ipni-advertisement-fetcher-job.service.ts

```diff
@@ -0,0 +1,142 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import {
+  HealthCheckError,
+  HealthIndicator,
+  HealthIndicatorResult,
+} from '@nestjs/terminus';
+import { PrismaService } from '../../db/prisma.service';
+import { CidContactService } from '../../service/cid-contact/cid-contact.service';
+import {
+  IPNIAdvertisement,
+  IPNIProvider,
+} from '../../service/cid-contact/types.cid-contact';
+
+@Injectable()
+export class IpniAdvertisementFetcherJobService extends HealthIndicator {
+  private readonly logger = new Logger(IpniAdvertisementFetcherJobService.name);
+  private healthy = true;
+  private jobInProgress = false;
+
+  constructor(
+    private readonly prismaService: PrismaService,
+    private readonly cidContactService: CidContactService,
+  ) {
+    super();
+  }
+
+  async isHealthy(): Promise<HealthIndicatorResult> {
+    const result = this.getStatus(
+      'ipni-advertisement-fetcher',
+      this.healthy,
+      {},
+    );
+
+    if (this.healthy) return result;
+    throw new HealthCheckError('Healthcheck failed', result);
+  }
+
+  @Cron(CronExpression.EVERY_DAY_AT_8PM)
+  async runIPNIAdvertisementFetcherJob() {
+    if (!this.jobInProgress) {
+      this.jobInProgress = true;
+
+      try {
+        this.logger.log('Starting IPNI Advertisement Fetcher job');
+        this.healthy = true;
+
+        await this._runIPNIAdvertisementFetcherJob();
+
+        this.logger.log(`Finishing IPNI Advertisement Fetcher job`);
+      } catch (err) {
+        this.healthy = false;
+        this.logger.error(
+          `Error while running IPNI Advertisement Fetcher job: ${err}`,
+          err.stack,
+        );
+      } finally {
+        this.jobInProgress = false;
+      }
+    } else {
+      this.logger.warn(
+        'IPNI Advertisement Fetcher job is already in progress - skipping next execution',
+      );
+    }
+  }
+
+  private async _runIPNIAdvertisementFetcherJob() {
+    // TODO ll error handling, count fails, success, prometheus metrics
+
+    const providers = await this.cidContactService.getIPNIProviders();
+
+    for (const [i, provider] of providers.entries()) {
```

Comment:
We are waiting for the result before we start the next one, 
How many providers are there ?
why are we not using the `Promise.allSettled()` ?

```suggestion
 const fetchPromises = providers.map(async (provider, i) => {
    try {
    ...
         await this.fetchAndStoreAdvertisementsByProvider(provider);
    } catch (err) {
    ....
    }
  });

  await Promise.allSettled(fetchPromises);
  ```
  
  I saw there are other places to implement it (clients and allocators report) to reduce the time, I think we will see the result of this implementation on the charts :smiley:

Reply:
This might cause problem with socket number limit / open connections limit, but it's a good suggestion. I'll test this locally and if it works, we'll change that in next PR.

Reply:
ok